src/popsift/s_orientation.cu

```diff
@@ -55,6 +60,7 @@ inline float compute_angle( int bin, float hc, float hn, float hp )
 /*
  * Histogram smoothing helper
  */
+#ifdef WITH_VLFEAT_SMOOTHING
```

Comment:
it would be better to use `POPSIFT_WITH_VLFEAT_SMOOTHING` or `POPSIFT_USE_VLFEAT_SMOOTHING` to avoid conflict and for consistency through repositories.

Also, is this supposed to be an option the user can choose or is it gonna be a kind of debugging or dev-only feature? 
In the first case, we should then also add the option in cmake and add the definition at compile time

```cmake
option(PopSift_USE_VLFEAT_SMOOTHING "Use VLFeat smoothing." ON)

...

# set the variable
if(PopSift_USE_VLFEAT_SMOOTHING)
  set(POPSIFT_USE_VLFEAT_SMOOTHING   1)
else()
  set(POPSIFT_USE_VLFEAT_SMOOTHING   0)
endif()

...

# add also the message in the final summary
message("******************************************")
message("Building configuration:\n")
message(STATUS "PopSift version: " ${PROJECT_VERSION})
message(STATUS "Build type: " ${CMAKE_BUILD_TYPE})
...
message(STATUS "Using VLFeat smoothing: " ${POPSIFT_USE_VLFEAT_SMOOTHING})
```

and add the entry in `cmake/sift_config.h`

```c++
#define POPSIFT_USE_VLFEAT_SMOOTHING()            @POPSIFT_USE_VLFEAT_SMOOTHING@
```

and `#include <popsift/sift_config.h>` wherever it is used

Reply:
I think we can remove the alternative and keep only VLFEAT_SMOOTHING. This is from the original development and the other setting has never been used.